src/software/backend/wifi_backend.cpp

```diff
@@ -8,15 +8,20 @@
 
 const std::string WifiBackend::name = "wifi";
 
-WifiBackend::WifiBackend()
-    : network_input(SSL_VISION_DEFAULT_MULTICAST_ADDRESS, SSL_VISION_MULTICAST_PORT,
-                    SSL_GAMECONTROLLER_MULTICAST_ADDRESS,
-                    SSL_GAMECONTROLLER_MULTICAST_PORT,
+WifiBackend::WifiBackend(std::shared_ptr<const NetworkConfig> network_config)
+    : network_config(network_config),
+      network_input(network_config->VisionIPv4Address()->value(),
+                    network_config->VisionPort()->value(),
+                    network_config->GamecontrollerIPv4Address()->value(),
+                    network_config->GamecontrollerPort()->value(),
                     boost::bind(&WifiBackend::receiveWorld, this, _1),
-                    DynamicParameters->getAIControlConfig()->getRefboxConfig(),
-                    DynamicParameters->getCameraConfig()),
+                    DynamicParameters->getSensorFusionConfig()),
       ssl_proto_client(boost::bind(&Backend::receiveSSLWrapperPacket, this, _1),
-                       boost::bind(&Backend::receiveSSLReferee, this, _1))
+                       boost::bind(&Backend::receiveSSLReferee, this, _1),
+                       network_config->VisionIPv4Address()->value(),
+                       network_config->VisionPort()->value(),
+                       network_config->GamecontrollerIPv4Address()->value(),
+                       network_config->GamecontrollerPort()->value())
```

Comment:
1. Why pass in individual values? Why not pass in a whole config?

Reply:
The Network.yaml config contains values that wouldn't be used, and I thought this was a low-enough level there wasn't a huge benefit in passing the config al the way down. I don't want to "overuse" them and put them too deep if not needed.

Reply:
this bit of code will need to be duplicated in almost every backend we make, so I'd like this to look tidier. For example, we could make an SSLProtoCommunicationConfig

Reply:
done as suggested.